operator/builtin/input/file/config.go

```diff
@@ -117,6 +119,24 @@ func (c InputConfig) Build(context operator.BuildContext) ([]operator.Operator,
 		return nil, fmt.Errorf("invalid start_at location '%s'", c.StartAt)
 	}
 
+	var labelRegex *regexp.Regexp
+	if c.LabelRegex != "" {
+		r, err := regexp.Compile(c.LabelRegex)
+		if err != nil {
+			return nil, fmt.Errorf("compiling regex: %s", err)
+		}
+
+		keys := r.SubexpNames()
+		// keys[0] is always the empty string
+		if len(keys) != 3 {
+			return nil, fmt.Errorf("label_regex must contain two capture groups named 'key' and 'value'")
+		}
+		if keys[1] != "key" || keys[2] != "value" {
```

Comment:
Could these be in reverse order?

Reply:
I think this way avoids indexing out of range, but I think we can combine them (just pushed a commit)
```
		if len(keys) != 3 || keys[1] != "key" || keys[2] != "value" {
			return nil, fmt.Errorf("label_regex must contain two capture groups named 'key' and 'value'")
		}
```

Reply:
Putting this another way: What if the user needs to specify a regex where value is captured before regex? I know this is unlikely, but it seems silly not to support it.